### PR TITLE
Thread pool and asynchronous HTTP request update (0.10.1)

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
Introduced changes:
- `HyperdriveAsyncClient.get_async` method based on `twisted.web.client.Agent`
- `HyperdriveResourceManager.pull` makes use of `HyperdriveAsyncClient.get_async` instead of `async_run` and `HyperdriveClient.get_file`
- macOS: `kqueue`-based reactor installed on macOS
- reactor thread pool's MAX size increased to 20

Nearing thread pool exhaustion, some HyperG API requests were never fired. In this PR, async API requests are handled by Twisted.

~TODO:~
For future use:
- `HyperdriveAsyncClient.add_async` method for adding files based on `twisted.web.client.Agent`

### CODE REVIEW

**Comment by mfranciszkiewicz** [comment]
Closing in favor of #1826

**Comment by mfranciszkiewicz** [comment]
Reopened, changes will be merged with the `b0.10.1` branch.

File: tests/golem/resource/test_resourceclient.py [hunk 1]
```diff
@@ -11,41 +14,130 @@
...
             def func():
-                nonlocal counter
-                counter += 1
-                raise ArithmeticError
+                self.state.increment()
+                raise ArithmeticError()
 
             self.handler._retry(func, raise_exc=False)
 
         # Exception was raised on first retry
```
Comment (maaktweluit) [inline tests/golem/resource/test_resourceclient.py:59]: does this comment still belong here?

**Review by maaktweluit — COMMENTED** [review]
LGTM!

One small comment, will approve when the other PR is merged.

File: tests/golem/resource/test_resourceclient.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply tests/golem/resource/test_resourceclient.py:59]: The test is still the same - an exception was raised on first retry. The exception is not on the "retry" list, so we stop the execution.

**Review by Elfoniok — APPROVED** [review]

**Review by badb — APPROVED** [review]

**Pull request merged**